### PR TITLE
feat[*]: Adding analytics events

### PR DESCRIPTION
**Pull request by falloutfox**
## Description:
Adding various analytics events for Sport1 app.
### Checklist for this pull request
- [x] PR is scoped to one task
- [ ] Tests included

- One of:
  - [x] The PR is not making any UI change
  - [ ] The PR is making a UI change
    - [ ] Screenshots included

 ## QA:
  - Required test cases:
  - Which apps this change/fix should be tested on:

Thank you!

### CODE REVIEW

**Comment by YoniOsteen** [comment]
Can you please share with me the requirements document for events and properties that you added?

Please note we have best practices for documentation on naming analytic events, which can be found [here](https://developer.applicaster.com/analytics/client_side_api/client_side_api.html#best-practices).

I also recommend looking at the documentation on player analytics [here](https://developer.applicaster.com/player/player-analytics.html) in order to stay consistent with other player plugins.

**Comment by YoniOsteen** [comment]
> Please don't use ApplicasterSDK for this
> you have ZAAppConnector sharedInstance analyticsDelegate
> 
> look into `ZAAppDelegateConnectorAnalyticsProtocol` which is the implemented protocol.

To Liviu's point above, you can find the documentation on sending analytics events [here](https://developer.applicaster.com/analytics/client_side_api/client_side_api.html).

**Comment by falloutfox** [comment]
Android PR aligning event names https://github.com/applicaster/JWPlayer-Plugin-Android/pull/17

**Comment by YoniOsteen** [comment]
Dev review still required of course, but looks good to me from a product perspective 💯

File: JWPlayerPlugin/JWPlayerViewController.m [hunk 1]
```diff
@@ -335,4 +345,62 @@ - (void)onControlBarVisible:(JWEvent<JWControlsEvent> *)event {
...
+    [[[ZAAppConnector sharedInstance] analyticsDelegate] trackEventWithName:@"Percentage Watched Of VOD"
+                                                                 parameters:extensions];
+}
+
+-(void)onSeek:(JWEvent<JWSeekEvent> *)event {
+    self.trackedPercentage = 0;
+}
+
+-(void)onAdPlay:(JWAdEvent<JWAdStateChangeEvent> *)event {
+    [extensions setObject:@"start"
```
Comment (guysarfatty) [inline JWPlayerPlugin/JWPlayerViewController.m]: Capital "S" on "Start"

File: JWPlayerPlugin/JWPlayerViewController.m [hunk 2]
```diff
@@ -335,4 +345,62 @@ - (void)onControlBarVisible:(JWEvent<JWControlsEvent> *)event {
...
+                   forKey:@"advertisement_position"];
+    [[[ZAAppConnector sharedInstance] analyticsDelegate] trackEventWithName:@"Watch Video Advertisement"
+                                                                 parameters:extentions];
+}
+
+-(void)onAdPause:(JWAdEvent<JWAdStateChangeEvent> *)event {
+    [extensions setObject:@"pause"
```
Comment (guysarfatty) [inline JWPlayerPlugin/JWPlayerViewController.m]: Capital "P"

File: JWPlayerPlugin/JWPlayerViewController.m [hunk 3]
```diff
@@ -335,4 +345,62 @@ - (void)onControlBarVisible:(JWEvent<JWControlsEvent> *)event {
...
+                   forKey:@"advertisement_position"];
+    [[[ZAAppConnector sharedInstance] analyticsDelegate] trackEventWithName:@"Watch Video Advertisement"
+                                                                 parameters:self.extensionsDictionary];
+}
+
+-(void)onAdComplete:(JWAdEvent<JWAdDetailEvent> *)event {
+    [extensions setObject:@"end"
```
Comment (guysarfatty) [inline JWPlayerPlugin/JWPlayerViewController.m]: Capital "E"

File: JWPlayerPlugin/JWPlayerViewController.m [hunk 4]
```diff
@@ -335,4 +345,62 @@ - (void)onControlBarVisible:(JWEvent<JWControlsEvent> *)event {
...
+        self.trackedPercentage = 75;
+    } else if (per >= 95.0 && per < 96.0 && self.trackedPercentage < 95) {
+        self.trackedPercentage = 95;
+    } else {
+        return;
+    }
+    NSMutableDictionary *extensions = self.extensionsDictionary.mutableCopy;
+    [extensions setObject:[NSNumber numberWithDouble:self.trackedPercentage]
+                   forKey:@"percentage"];
+    [[[ZAAppConnector sharedInstance] analyticsDelegate] trackEventWithName:@"Percentage Watched Of VOD"
```
Comment (guysarfatty) [inline JWPlayerPlugin/JWPlayerViewController.m]: Event name should be: "Watch VOD percentage" to be aligned with Android

**Pull request merged**